### PR TITLE
Add support for map type configurable variables

### LINKED ISSUE #30389: Support map<anydata> for Configurables
The configurable variables must support `map<anydata>` type through TOML syntax.
This issue  describes the map of constraints with a specific type. 
Eg. - 
```ballerina
type Person record {|
    int id;
    string name;
|};
configurable map<int> & readonly numbers = ?;
configurable map<string[][]> & readonly stringSet = ?;
configurable map<Person> & readonly person = ?;
```
These variables can be configured with following TOML syntax

```toml
[numbers]
one = 1
two = 2
three = 3
four = 4

[stringSet]
set1 = [["a","aa","aaa"], ["b","bb","bbb"],["c","cc","ccc"]]
set2 = [["d","dd","ddd"], ["e","ee","eee"],["f","ff","fff"]]
set3 = [["g","gg","ggg"], ["h","hh","hhh"],["i","ii","iii"]]

[person]
person1.id = 11
person1.name = "Tom"
person2.id = 22
person2.name = "John"
person3.id = 33
person3.name = "Jack"

```

In similar ways, 
we need to support 
 - `map<anydata>[]`
 - `table<map<anydata>>`

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

Fixes #30389

## Approach
Toml syntax support map typed variables through TOML tables. 

```ballerina
configurable map<int> & readonly numbers = ?;
configurable map<int>[] & readonly intMapArr = ?;
configurable table<map<string>> & readonly stringMapTable = ?;
```

can be provided as 
```toml
[numbers]
one = 1
two = 2
three = 3
four = 4

[[intMapArr]]
int1 = 34
int2 = 78

[[intMapArr]]
int3 = 17
int4 = 92

[[stringMapTable]]
name = "Tom"
city = "Colombo"

[[stringMapTable]]
name = "Harry"
city = "Kandy"
```

## Remarks
Fixing ambiguities is tracked in #30571 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java [hunk 1]
```diff
@@ -259,18 +259,89 @@ public boolean hasConfigs() {
...
     @Override
     public Optional<BTable<BString, Object>> getAsTableAndMark(Module module, VariableKey key) {
-        Type effectiveType = ((BIntersectionType) key.type).getConstituentTypes().get(0);
+        TableType tableType = (TableType) ((BIntersectionType) key.type).getEffectiveType();
+        Type constraintType = tableType.getConstrainedType();
+        if (constraintType.getTag() == TypeTags.INTERSECTION_TAG) {
+            constraintType = ((IntersectionType) constraintType).getEffectiveType();
+        }
+        if (constraintType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+            tableType = (TableType) ((BIntersectionType) key.type).getConstituentTypes().get(0);
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java:339]: Do we have tests for map types defined in another module?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java [hunk 2]
```diff
@@ -259,18 +259,89 @@ public boolean hasConfigs() {
...
+                    retrieveTypedValue(field.getValue(), variableName + "." + fieldName, mapType.getConstrainedType());
+            keyValueEntries[count] =
+                    new MappingInitialValueEntry.KeyValueEntry(StringUtils.fromString(fieldName), value);
+            count++;
+
+        }
+        return ValueCreator.createMapValue(mapType, keyValueEntries);
+    }
+
+    private Object retrieveTypedValue(TomlNode tomlValue, String variableName, Type type) {
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java:306]: Shall we rename this method to "retrieveValue" or "retrieveBalValue"

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java (hunk 1 above)
Comment (warunalakshitha) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java:339]: Also defined in another project module?

Comment (HindujaB) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java:339]: added in last commit.

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**